Sources/PT.PM.AntlrUtils/LightToken.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using Antlr4.Runtime;
+using PT.PM.Common;
+using PT.PM.Common.Files;
+
+namespace PT.PM.AntlrUtils
+{
+    // TODO: should store TextSpan instead of StartIndex and StopIndex
+    public readonly struct LightToken : IToken
+    {
+        private readonly LightInputStream inputStream;
+        private readonly short type;
+        private readonly short channel;
+
+        public int StartIndex { get; }
+
+        public int StopIndex { get; }
+
+        public int TokenIndex { get; }
+
+        public LightToken(LightInputStream inputStream, int type, int channel, int index, int start, int stop)
+        {
+            this.inputStream = inputStream;
+            this.type = (short) type;
```

Comment:
What's the point in accepting `int`s and truncating them to `short`s?

Reply:
Memory conserving. Now we have `short type` and `short channel`. Together they take 4 bytes instead of 8. It makes sense for huge projects.